server/public/plugin/plugintest/example_reattach_test.go

```diff
@@ -0,0 +1,85 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package plugintest_test
+
+import (
+	"context"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+
+	goPlugin "github.com/hashicorp/go-plugin"
+	"github.com/mattermost/mattermost/server/public/model"
+	"github.com/mattermost/mattermost/server/public/plugin"
+)
+
+type UnitTestedPlugin struct {
+	plugin.MattermostPlugin
+}
+
+// This example demonstrates a plugin that's launched during a unit test and reattached to an
```

Comment:
0/5 on whether this example is useful or not -- see https://github.com/mattermost/mattermost-plugin-msteams/pull/563 for how we productize this exact code in a helper function.

Reply:
I quite like the example, as it should _a_ way to implement e2e tests. It also makes it easy for the community to pick up e2e testing!